qemu/tests/vdpa_sim_blk_test.py

```diff
@@ -0,0 +1,149 @@
+"""VDPA simulator blk test"""
+
+from avocado.utils import process
+
+from provider.block_devices_plug import BlockDevicesPlug
+from provider.vdpa_sim_blk_utils import VDPABlkSimulatorTest
+from virttest import env_process, utils_disk, utils_misc
+from virttest.utils_misc import get_linux_drive_path
+
+
+def run(test, params, env):
+    """
+    VDPA simulator blk test. It tests multiple scenarios:
+    multi-disks,hotplug-unplug,and disk feature (discard)
+    Multi-disks steps:
+        1) Setup VDPA simulator ENV and add block devices.
+        2) Simple IO test on host
+        3) Boot VM with multi vdpa-blk disks
+        4) Verify disk in guest
+        5) Destroy VM
+        6) Destroy vdpa-blk disk on host
+        7. Destroy VDPA simulator ENV
+    Hotplug-unplug test steps:
+        1) Setup VDPA simulator ENV and add block devices.
+        2) Simple IO test on host
+        3) Boot VM
+        4) Hotplug the vdpa-blk disks
+        5) Verify disks in guest
+        6) Unplug those vdpa-blk disks
+        7) Destroy VM
+        8) Destroy vdpa-blk disk on host
+        9. Destroy VDPA simulator ENV
+    Discard feature test steps:
+        1) Setup VDPA simulator ENV and add block devices.
+        2) Simple IO test on host
+        3) blkdiscard operation on vdpa-blk disks
+        4) Destroy vdpa-blk disk on host
+        5. Destroy VDPA simulator ENV
+    """
+
+    def _setup_vdpa_disks():
+        for img in vdpa_blk_images:
+            dev = vdpa_blk_test.add_vdpa_blk_dev(img)
+            vdpa_blk_info[img] = "/dev/%s" % dev
+            params["image_name_%s" % img] = vdpa_blk_info[img]
+            cmd = host_cmd.format(dev)
+            process.run(cmd, shell=True)
+
+    def _cleanup_vdpa_disks():
+        for img in vdpa_blk_images:
+            vdpa_blk_test.remove_vdpa_blk_dev(img)
+
+    def _get_window_disk_index_by_serial(serial):
```

Comment:
FYI, the function "get_disk_props_by_serial_number" from vt should meet the requirement.
virttest.utils_windows.drive.py --> get_disk_props_by_serial_number(session, serial_number, props)

Reply:
updated, thanks.